### PR TITLE
Add continuous security analysis

### PR DESCRIPTION
**Pull request by 0ortmann**
LGTM integrates with github checks (see also this PR's checks). However, so far it does not analyse C++. That is because we need a custom extraction step, which is just introduced with this branch - and hence not enabled on the main repository.
This PR will enable C++ analysis on master.

### CODE REVIEW

File: lgtm.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - ln -s /usr/bin/g++-8 $LGTM_WORKSPACE/latest-gcc-symlinks/g++
+      - ln -s /usr/bin/gcc-8 $LGTM_WORKSPACE/latest-gcc-symlinks/gcc
+      - export BUILD_TYPE=Release
+      - export GIT=true
+      - export PATH=$LGTM_WORKSPACE/latest-gcc-symlinks:$PATH
+      - export PREFIX=/opt/tenzir
+    configure:
+      command:
+        - pip3 install cmake
+        - /opt/work/.local/lib/python3.6/site-packages/cmake/data/bin/cmake --version
```
Comment (mavam) [inline lgtm.yaml]: Those explicit paths seem to be brittle when it comes to version changes. Is there perhaps a cleaner approach based on virtual envs? (This assumes that the environment can be mutated between individual steps.)

Comment (0ortmann) [reply lgtm.yaml]: @mavam We could resolve the site package location dynamically, such as `$(python -m site --user-site)/cmake/data/bin/cmake --version`

**Comment by 0ortmann** [comment]
LGTM integrates with github checks (see also this PR's checks). However, so far it does not analyse C++. That is because we need a custom extraction step, which is just introduced with this branch - and hence not enabled on the main repository.
This PR will enable C++ analysis on master.

**Comment by 0ortmann** [comment]
This is the direct link to a successful LGTM run with C++ extraction
https://lgtm.com/logs/3a95781801eace2193259800a06bdf29478085ac/lang:cpp

File: lgtm.yaml (hunk 1 above)
Comment (mavam) [reply lgtm.yaml]: Seems better, albeit still noisy.

File: lgtm.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  cpp:
+    prepare:
+      packages:
+        - libssl-dev
+        - python-pip
+    after_prepare:
+      - mkdir -p $LGTM_WORKSPACE/latest-gcc-symlinks
+      - ln -s /usr/bin/g++-8 $LGTM_WORKSPACE/latest-gcc-symlinks/g++
+      - ln -s /usr/bin/gcc-8 $LGTM_WORKSPACE/latest-gcc-symlinks/gcc
+      - export BUILD_TYPE=Release
```
Comment (dominiklohmann) [inline lgtm.yaml]: Should this be Debug so it looks at all the codepaths that are hidden behind `#if VAST_LOG_LEVEL >= VAST_LOG_LEVEL_DEBUG` as well?

Comment (0ortmann) [reply lgtm.yaml]: As discussed with the team, we'll go with `Release` for now, as there is no important logic to be conditionally compiled in `Debug` builds. `Debug` affects mostly logging behavior.

**Pull request merged**